kornia/contrib/kmeans.py

```diff
@@ -0,0 +1,155 @@
+# based on https://github.com/subhadarship/kmeans_pytorch
+
+import torch
+
+
+class KMeans:
+    def __init__(
+        self,
+        num_clusters: int,
+        cluster_centers: torch.Tensor,
+        tolerance: float = 10e-4,
+        max_iterations: int = 0,
+        device: torch.device = torch.device('cpu'),
+        seed: int = None,
+    ) -> None:
+        if num_clusters == 0:
+            raise ValueError("num_clusters can't be 0")
+
+        self.num_clusters = num_clusters
+        self.cluster_centers = cluster_centers
+        self.tolerance = tolerance
+        self.max_iterations = max_iterations
+        self.device = device
+
+        self.final_cluster_assignments = None
+        self.final_cluster_centers = None
+
+        if seed is not None:
+            torch.manual_seed(seed)
```

Comment:
@johnnv1 we should have a wrapper around this I guess

Reply:
agree, should seed python random too